### PR TITLE
ENH: unzip support for download

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #103

Adds support to unzip files after download for zip files. Users can now specify a 'zip_method' as part of the download tags, which will send the zipped file to a temporary directory and extract it to the archival location.

Current version supports 'zip' only, but could be expanded for other file types later.

# Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested with pysatModels at https://github.com/pysat/pysatModels/pull/132

**Test Configuration**:
* Operating system: Monterrey
* Version number: Python 3.10.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
Needs a test for the new bits of code. The only routine that uses this is in pysatModels. Potentially we can create a test object with a few zipped files hosted here.

**Comment by jklenzing** [comment]
Removed the gz support for now since the TIMED datasets have other problems. Can be expanded in the future.

**Comment by jklenzing** [comment]
Documented need for tests and possible solutions as #213.

**Comment by jklenzing** [comment]
Coverage change consistent with new lines of code.

File: pysatNASA/instruments/methods/cdaweb.py [hunk 1]
```diff
@@ -526,6 +527,13 @@ def download(date_array, data_path, tag='', inst_id='', supported_tags=None,
                                      start=date_array[0],
                                      stop=date_array[-1])
 
+    # Create temproary directory if files need to be unzipped.
```
Comment (aburrell) [inline pysatNASA/instruments/methods/cdaweb.py:530]: Can you add a comment above line 518 that summarizes briefly what's in `inst_dict`?  Doesn't need to list all possible keys, just say in a sentence what it is.

File: pysatNASA/instruments/methods/cdaweb.py [hunk 2]
```diff
@@ -568,6 +577,52 @@ def download(date_array, data_path, tag='', inst_id='', supported_tags=None,
...
+    with open(dl_fname, 'wb') as open_f:
+        open_f.write(remote_file)
+
+    # Unzip and move the files from the temporary directory.
+    if zip_method == 'zip':
+        with zipfile.ZipFile(dl_fname, 'r') as open_zip:
+            open_zip.extractall(data_path)
+
+    elif zip_method is not None:
+        logger.warning('{:} is not a recognized zip method'.format(zip_method))
```
Comment (aburrell) [inline pysatNASA/instruments/methods/cdaweb.py:624]: Recommend adding a test for this warning.  Also creating an issue for the future tests, if it hasn't been done already.

Comment (jklenzing) [reply pysatNASA/instruments/methods/cdaweb.py:625]: documented as #213

**Review by aburrell — APPROVED** [review]

**Pull request merged**